test/performance/odp_sched_pktio.c

```diff
@@ -238,6 +238,24 @@ static int worker_thread_direct(void *arg)
 			test_global->worker_stat[worker_id].rx_pkt += num_pkt;
 	}
 
+	/*
+	 * Free prefetched packets before exiting worker thread as
+	 * such packets can block main thread cleanup cause buffer leak.
+	 */
+	odp_schedule_pause();
+	while (1) {
+		odp_event_t  ev;
+
+		ev = odp_schedule(NULL, ODP_SCHED_NO_WAIT);
+		if (ev == ODP_EVENT_INVALID)
+			break;
+		odp_event_free(ev);
+	}
+
+	/* Flush all writes */
+	odp_mb_full();
```

Comment:
I wonder if and why this is needed. I would have thought thread exit followed by odph_odpthreads_join() ensures the needed memory synchronization between the threads.

Reply:
Ack, Will remove this in V2. Join should be doing it.